### PR TITLE
Refs EN-1590

### PR DESCRIPTION
**Pull request by rozap**
* Check the contents of the archive before becoming readable
* Added shp and dbf missing fixtures and tests

### CODE REVIEW

File: es6-lib/decoders/shapefile.js [hunk 1]
```diff
@@ -164,18 +171,35 @@ class Shapefile extends Duplex {
     return emitter;
   }
 
+  _hasRequiredComponents(components) {
+    var groups = this._groupComponents(components);
+    return _.reduce(groups, (errors, [shp, _prj, dbf]) => {
+      //if the shp or dbf are undefined, then record the error. prj
+      //is technically optional
+      if(!shp) {
```
Comment (catstavi) [inline es6-lib/decoders/shapefile.js]: Nit for spacing here and on 182, unless there's a convention I'm unaware of.

File: es6-lib/decoders/shapefile.js [hunk 2]
```diff
@@ -164,18 +171,35 @@ class Shapefile extends Duplex {
...
+    return _.zip(shps, projs, dbfs);
   }
 
   _startPushing() {
     this._isPushing = true;
 
     var groups = this._groupComponents(this._components);
-    async.mapSeries(groups, ([shp, proj], cb) => {
+
+    async.mapSeries(groups, ([shp, proj, _dbf], cb) => {
```
Comment (catstavi) [inline es6-lib/decoders/shapefile.js]: What's the significance of the naming with an underscore before dbf here (and on _prj on line 176)?

Comment (rozap) [inline es6-lib/decoders/shapefile.js]: it means that the variable is unused, but it's there to make the shape of the datum clear

**Comment by catstavi** [comment]
If you fix the space-nits and figure out the build issues, I say :ship:

**Comment by rozap** [comment]
jenkins is just being a butt

**Comment by rozap** [comment]
but i will wait

**Comment by rozap** [comment]
it's ok jenkins i'll just wait here

**Comment by rozap** [comment]
take your time, friend

**Pull request merged**